### PR TITLE
Upgrade facebook client to 3.2

### PR DESCRIPTION
**Pull request by philmcmahon**
## What does this change?
We have been using Facebook API v2.8, which expires 18th April (today!) - this updates it to version 3.2.

## What is the value of this and can you measure success?
Without this change, share counts will stop working on the website.

### Tested

- [ ] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: applications/app/services/FacebookGraphApi.scala [hunk 1]
```diff
@@ -9,20 +9,18 @@ import play.api.libs.ws.{WSClient, WSResponse}
...
   implicit val jsonResponse = Json.reads[URLResponse]
 }
 
-case class URLResponse(id: String, share: ShareObject)
-case class ShareObject(share_count: Int)
+case class Engagement(share_count: Int)
+case class URLResponse(id: String, engagement: Engagement)
 
 class FacebookGraphApiClient(wsClient: WSClient) extends implicits.WSRequests with Logging {
-  val apiRootUrl = s"https://graph.facebook.com/v${Configuration.facebook.graphApi.version}/"
```
Comment (rtyley) [inline applications/app/services/FacebookGraphApi.scala]: Guess this `/`was superfluous?

**Review by rtyley — APPROVED** [review]
Looks  great to me! The similar PR for Ophan was https://github.com/guardian/ophan/pull/3268

**Comment by rtyley** [comment]
I fixed up the tests with 614fd40 and deployed to the CODE environment:

### Before (PROD environment)

https://www.theguardian.com/commentisfree/2015/may/27/twitter-police-enable-hate-speech

image

### After (CODE environment)

https://m.code.dev-theguardian.com/commentisfree/2015/may/27/twitter-police-enable-hate-speech
image

...so, that looks better!

**Review by shtukas — APPROVED** [review]

**Comment by rtyley** [comment]
Correct values are now showing again! Thank you for reviewing at this late hour @shtukas !

image

**Pull request merged**